enhancements/installer/vsphere-ipi-multiple-datacenter-clusters.md

```diff
@@ -0,0 +1,964 @@
+including `Datacenter`, `ComputeCluster`, `Hosts`, `Networks` and a `Datastore`.
+
+`DeploymentZones` will define the association between a FailureDomain, vcenter and
+PlacementConstraints - which include Resource Pools and Folders.
+
+The existing platform spec vcenter parameters will be deprecated
+but _not_ removed or remove support for using those parameters.
+
+```golang
+// VSpherePlatformFailureDomainType is the name of the failure domain type.
+// There are two defined failure domains currently, Datacenter and ComputeCluster.
+// Each represents a vCenter object type within a vSphere environment.
+// +kubebuilder:validation:Enum="";HostGroup;Datacenter;ComputeCluster
+type FailureDomainType string
+
+const (
+    // HostGroup as a type allows the use of a group of ESXi hosts
+    // to be represented as a failure domain zone. When using this
+    // case it is expected that region would be a cluster.
+    // HostGroups within vCenter must be preconfigured and
+    // assigned in the topology.
+    HostGroupFailureDomain      VSpherePlatformFailureDomainType = "HostGroup"
+
+    // ComputeCluster failure domain can either be a zone or region.
+    // The vCenter cluster is required to preconfigured and
+    // assigned in the topology.
+    ComputeClusterFailureDomain VSpherePlatformFailureDomainType = "ComputeCluster"
+
+    // Datacenter failure domain can be only a region. The vcenter
+    // datacenter is required to be preconfigred and assigned
+    // in the topology. If used the zone would be of type ComputeCluster.
+    DatacenterFailureDomain     VSpherePlatformFailureDomainType = "Datacenter"
+)
+
+// FailureDomainSpec holds the region and zone failure domain and
+// the vCenter topology of that failure domain.
+type FailureDomainSpec struct {
+  // name defines the name of the FailureDomainSpec
+  // This name is abritary but will be used
+  // in VSpherePlatformDeploymentZone for association.
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=256
+  Name string `json:"name"`
+
+  // region defines a FailureDomain which
+  // includes the name of the vCenter tag, the failure domain type
+  // and the name of the vCenter tag category.
+  // +kubebuilder:validation:Required
+  Region FailureDomain `json:"region"`
+
+  // zone defines a VSpherePlatformFailureDomain which
+  // includes the name of the vCenter tag, the failure domain type
+  // and the name of the vCenter tag category.
+  // +kubebuilder:validation:Required
+  Zone FailureDomain `json:"zone"`
+
+  // Topology describes a given failure domain using vSphere constructs
+  // +kubebuilder:validation:Required
+  Topology Topology `json:"topology"`
+}
+
+// FailureDomain holds the name of the associated tag, the type
+// of the failure domain, and the vCenter tag category associated with this
+// failure domain.
+type FailureDomain struct {
```

Comment:
What is the difference between `FailureDomain` here and `VSpherePlatformFailureDomain` specified above - https://github.com/openshift/enhancements/pull/918/files#diff-1abad80739bb01679e9998c2b1ef428495371d5b7826a0585532a5cefb4762daR112 ?